### PR TITLE
[Filter/Sub] preserved name

### PR DESCRIPTION
**Pull request by jaeyun-jung**
do not allow 'auto' as subplugin name.

Signed-off-by: Jaeyun <jy1210.jung@samsung.com>

### CODE REVIEW

**Comment by jaeyun-jung** [comment]
Discussion: consider filter-fw auto detection from @gichan-jang

File: api/android/api/src/androidTest/java/org/nnsuite/nnstreamer/APITestCustomFilter.java [hunk 1]
```diff
@@ -311,4 +311,26 @@ public TensorsData invoke(TensorsData in) {
             /* expected */
         }
     }
+
+    @Test
+    public void testRegisterPreservedName_n() {
+        try {
+            CustomFilter.registerCustomFilter("auto",
```
Comment (myungjoo) [inline api/android/api/src/androidTest/java/org/nnsuite/nnstreamer/APITestCustomFilter.java]: Even if tensor-filter subplugin "auto" is prohibited, a custom plugin "auto" should be allowed. If it's customfilter "auto", its subplugin name will be "custom" or "custom-cpp" or "custom-easy" or so.

File: gst/nnstreamer/nnstreamer_subplugin.c [hunk 2]
```diff
@@ -184,6 +184,12 @@ register_subplugin (subpluginType type, const char *name, const void *data)
       return FALSE;
   }
 
+  /* check the sub-pugin name */
+  if (g_ascii_strcasecmp (name, "auto") == 0) {
+    g_critical ("Failed, the name %s is not allowed.", name);
+    return FALSE;
+  }
+
```
Comment (myungjoo) [inline gst/nnstreamer/nnstreamer_subplugin.c]: LGTM.

File: api/android/api/src/androidTest/java/org/nnsuite/nnstreamer/APITestCustomFilter.java (hunk 1 above)
Comment (jaeyun-jung) [reply api/android/api/src/androidTest/java/org/nnsuite/nnstreamer/APITestCustomFilter.java]: custom-filter in java is not a .so file (run without model file). It runs without so, and the fw name would be auto in this case.

Comment (myungjoo) [reply api/android/api/src/androidTest/java/org/nnsuite/nnstreamer/APITestCustomFilter.java]: So, it is a custom filter with "auto" as its custom filter name?

What if we have two java custom filters in a single pipeline, then?

Comment (jaeyun-jung) [reply api/android/api/src/androidTest/java/org/nnsuite/nnstreamer/APITestCustomFilter.java]: If developer needs two custom-filters, implement two interfaces with different names and register them.
```
CustomFilter.registerCustomFilter("custom1", new CustomFilter.CustomFilterCallback() { ..
CustomFilter.registerCustomFilter("custom2", new CustomFilter.CustomFilterCallback() { ..

pipeline = "... ! tensor_filter framework=custom1 ! tensor_filter framework=custom2 ! ..."
```

**Comment by jaeyun-jung** [comment]
2020-03-17 PR updated
- add test for preserved subplugin name

**Pull request merged**